[Le Quoc Huy] iP

jeremy-7007 opened the pull request:
# Nukem
> "Believe every advertisement you see on the Internet. They're all true." - Steve Jobs

Have the memory of an ADHD seagull? Nukem is here for you! It's,
* text-based (you nerd)
* so easy to learn my grandpa is using it _(RIP grandpa)_
* ~~FAST~~ ~~nay~~ _SUPER_ FAST to use (if you're a bot)

To experience the excitement,
1. download it from [here](https://www.reddit.com/r/YouFellForItFool/comments/cjlngm/you_fell_for_it_fool/?utm_source=share&utm_medium=web2x).
2. double click it.
3. add your tasks (figure out how yourself).
4. let it remember your tasks for you. That's all it's good for! 😉 👍 

And it is **FREE**! *(Please call customer support at 6666 6666 if you were charged for your version of Nukem)*

Features:
- [X] Managing tasks like a nagging mother
- [ ] Managing deadlines (coming soon once the engineers stop throwing bananas)
- [ ] Reminders (coming soon to computers near you)

#

If you're a Java programmer, you can do my job for me too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke("./Duke.txt").run();
    }
}
```

Commit:
Add Gradle support

jeremy-7007 committed:
Commit level 1

jeremy-7007 committed:
Commit level 2

jeremy-7007 committed:
Added class Task

jeremy-7007 committed:
Commit level 3

jeremy-7007 committed:
Commands show error instead of adding a new task. New level 3.

jeremy-7007 committed:
Support todo command

jeremy-7007 committed:
Commit level 4

jeremy-7007 committed:
Commit A-TextUiTesting

jeremy-7007 committed:
Commit level 5

jeremy-7007 committed:
Commit level 6

jeremy-7007 committed:
Create and read file

jeremy-7007 committed:
Write new task to file

jeremy-7007 committed:
Commit level 7

jeremy-7007 committed:
Commit level 8 to branch

jeremy-7007 committed:
Merge branch level 8, resolve conflicts

jeremy-7007 committed:
OOP WIP

jeremy-7007 committed:
Commit A-MoreOOP

jeremy-7007 committed:
Commit A-Packages

jeremy-7007 committed:
Commit A-JUnit

jeremy-7007 committed:
Commit A-Jar

jeremy-7007 committed:
Commit A-JavaDoc

jeremy-7007 committed:
Commit Level-9

jeremy-7007 committed:
Merge branch 'branch-A-JavaDoc'

jeremy-7007 committed:
Merge branch 'branch-Level-9'

jeremy-7007 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jeremy-7007 committed:
Commit A-Gradle

jeremy-7007 committed:
Commit A-Checkstyle

ErnestCuong commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+        return this.taskList.size();
+    }
+
+    public String searchTask(String searchWord) {
```
Perhaps you can add javadocs for this public method.

ErnestCuong commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,105 @@
+
+import java.util.Scanner;
+
+public class Ui {
```
Maybe you can consider adding javadocs for this class.

ErnestCuong commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,38 @@
+import duke.TaskList;
+import duke.Ui;
+
+public class AddCommand extends Command {
```
I think you might not have enough time before to add javadocs which is understandable. You can consider adding them for public classes and methods as soon as you can.

jeremy-7007 committed:
Almost finish JavaFX tutorial

jeremy-7007 committed:
Finish tutorial

jeremy-7007 committed:
Fix logic with commands and ui

ErnestCuong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,82 @@
+     *
+     * @return String representing whether the task is done.
+     */
+    public String getStatusIcon() {
```
I like this way of getting the mark for task done. It is convenient and can be added to string much more easily.

ErnestCuong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+//    @Override
```
Perhaps you can consider deleting unused pieces of code to declutter.

ErnestCuong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,123 @@
+                return new ListCommand();
+            } else {
+                throw new DukeException("Command 'list' does not need additional arguments");
+            }
```
I notice you did not use `break;`. Perhaps you can write comments explaining why it is not necessary to put it.

ErnestCuong commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,42 @@
+     * @param ui the Ui used during execution.
+     */
+    @Override
+    public void execute(TaskList taskList, Storage storage, Ui ui) {
```
I think perhaps you can indicate that this method throws `DukeException` by adding `throws DukeException` after the parameters.

ErnestCuong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,82 @@
+     *
+     * @param newDesc New description to give to the task.
+     */
+    public void changeDescription(String newDesc) {
```
Maybe can name this method as `changeDescriptionTo` ?

ErnestCuong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,123 @@
+     * @throws DukeException If input is not recognised or does not follow format.
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] wordArray = input.split(" ");
```
Maybe you can consider naming this as `charArray`? I am not too sure, you can correct me if I'm wrong.

ErnestCuong reviewed (commented):
Looks good to merge. Apart from a few missing javadocs, and possibly some minor coding problems, I think you have done very a great job.

jeremy-7007 committed:
Change images

jeremy-7007 committed:
Merge branch 'branch-Level-10'

jeremy-7007 committed:
Add assertions in several places

Assertions are not used in the code since most variable cases are
accounted already accounted for.

Adding assertions will make the code even more error-proof.

Let's add assertions at several points to make sure there can be no
error made by the program.

jeremy-7007 committed:
Improves code quality

Some methods and variables are long and does different things in
the same line or method.

Abstracting these out will make the code more readable and simple.

Let's abstract out these methods and variables to improve the code
quality and make it more readable.

jeremy-7007 committed:
Merge pull request #4 from jeremy-7007/branch-A-Assertions

Add assertions in several places

jeremy-7007 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	Duke.txt
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java

jeremy-7007 committed:
Merge pull request #5 from jeremy-7007/branch-A-CodeQuality

Improves code quality

jeremy-7007 committed:
Sets up Continuous Integration

jeremy-7007 committed:
Adds function to detect duplicate task when new task is added

jeremy-7007 committed:
Merge pull request #6 from jeremy-7007/branch-C-DetectDuplicates

Adds function to detect duplicate task when new task is added

jeremy-7007 committed:
Improve GUI slightly

jeremy-7007 committed:
Improve text box

jeremy-7007 committed:
Writes the user guide

jeremy-7007 committed:
Set theme jekyll-theme-hacker
